src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/Processor.java

```diff
@@ -56,6 +56,9 @@ protected void processParsedEnvelopeDocuments(
         List<Pdf> pdfs,
         CloudBlockBlob cloudBlockBlob
     ) {
+        if (envelope == null) {
+            return;
```

Comment:
why is envelope null? Looks like something that deserves a log

Reply:
It is logged. There's a separation in zip processor in this PR. There's a chance envelope can be null